.github/workflows/async-auto-label.yml

```diff
@@ -24,46 +28,68 @@ jobs:
           comment: ${{ github.event.comment.body }}
           requester: ${{ github.event.comment.user.login }}
           repo: ${{ github.event.repository.full_name }}
-          pr: ${{ github.event.issue.pull_request.number }}
+          pr: ${{ github.event.issue.number }}
           token: ${{ github.token }}
+          association: ${{ github.event.comment.author_association }}
         run: |
           import re
           import os
           import github
 
-          match = re.match(r'\+async-label\s+(.+)', os.environ['comment'])
-          assert match, f'could not parse comment: {os.environ["comment"]!r}'
-          labels = [label.strip() for label in match.group(1).split(',')]
-
+          accepted_associations = ['COLLABORATOR', 'CONTRIBUTOR', 'MEMBER', 'OWNER']
           repo = github.Github(os.environ['token']).get_repo(os.environ['repo'])
-          pr = repo.get_pull(os.environ['pr'])
+          pr = repo.get_pull(int(os.environ['pr']))
+          requester = os.environ['requester']
+
+          def modify_labels():
+            pattern = pattern = r'(!?\+label-async-\w+)'
+            matches = re.findall(pattern, os.environ['comment'])
```

Comment:
I see you changed the `+async-label label1, !label2` syntax to `+label-async-label1 !+label-async-label2`.

The syntax itself is a matter of taste of course, but would it make sense to process the new syntax e.g. only when it appears on its own line, or only at the start of a comment? Otherwise people might use it in a sentence and not expect the labels to actually be changed.

Reply:
Actually, I see that I made it much more complicated.
I will go back to your implementation. I think I had a misunderstanding during my implementation!

Reply:
I went back to the old syntax, see https://github.com/benedikt-voelkel/ali-bot/pull/3